nix/make-shell.nix

```diff
@@ -0,0 +1,45 @@
+{ repoRoot, pkgs, ... }:
+
+cabalProject:
+
+{
```

Comment:
I find the naming confusing here. "make X" is usually a function that makes an X, which this is not. I'm _guessing_ that it's the _arguments_ to the `mkShell` function, so I'd call it `mkShellArgs` or something.

It's also not a module? Or is it? Why is it getting the cabal project as an argument instead of accessing module settings? I'm generally confused.